app/controllers/jobs_controller.rb

```diff
@@ -1,15 +1,28 @@
 class JobsController < ApplicationController
+
+
   def index
-    @company = Company.find(params[:company_id])
-    @jobs = @company.jobs
+    if params[:sort]
+
+      @company = Company.find(params[:company_id])
+      @jobs = Job.where(city: params[:sort], company: @company)
+      @contact = Contact.new
+
+    else
+      @company = Company.find(params[:company_id])
+      @jobs = @company.jobs
+      @contact = Contact.new
+    end
   end
 
   def new
+    @categories = Category.all
     @company = Company.find(params[:company_id])
     @job = Job.new()
   end
 
   def create
+    @categories = Category.all
```

Comment:
dont need all the categories to create...the category_id should be coming through the params from the new form.